memory-viz/src/memory_model.ts

```diff
@@ -1145,4 +1204,96 @@ export class MemoryModel {
             this.svg.setAttribute("height", this.height.toString());
         }
     }
+
+    /**
+     * Add hover interactivity to the SVG on object IDs
+     */
+    setInteractivityScript(): void {
+        const idToObjectMapping = Object.fromEntries(this.idToObjectMap);
+
+        const script = `
+            function objectInteractivity() {
+                // Inject the id value to object id mapping into script
+                const idToObjectMap = ${JSON.stringify(idToObjectMapping)};
+                
+                function highlightObject(objectId) {
+                    const objectBox = document.getElementById(objectId);
+                    if (!objectBox) return;
+                    
+                    const bbox = objectBox.getBBox();
+                    const existingFill = objectBox.querySelector('path[fill]:not([fill="none"])');
```

Comment:
I mentioned this on Slack, but let's not worry about preserving fill explicitly like this. Instead, use a CSS class, and if necessary you can set the rule to `!important` to override any user-provided fill values.